libs/application/templates/marriage-conditions/src/lib/dataSchema.ts

```diff
@@ -40,55 +40,68 @@ export const dataSchema = z.object({
   ceremony: z
     .object({
       hasDate: z.string(),
-      withDate: z
+      date: z.string(),
+      period: z
         .object({
-          date: z.string(),
-          ceremonyPlace: z.string(),
-          office: z.string(),
-          society: z.string(),
-        })
-        .partial()
-        .refine(({ date }) => !!date, {
-          message: coreErrorMessages.defaultError.defaultMessage,
-          path: ['date'],
-        })
-        .refine(({ ceremonyPlace }) => !!ceremonyPlace, {
-          message: coreErrorMessages.defaultError.defaultMessage,
-          path: ['ceremonyPlace'],
+          dateFrom: z.string().optional(),
+          dateTil: z.string().optional(),
         })
         .refine(
-          ({ ceremonyPlace, office }) =>
-            ceremonyPlace === CeremonyPlaces.office ? !!office : true,
-          {
-            message: coreErrorMessages.defaultError.defaultMessage,
-            path: ['office'],
-          },
-        )
-        .refine(
-          ({ ceremonyPlace, society }) =>
-            ceremonyPlace === CeremonyPlaces.society ? !!society : true,
-          {
-            message: coreErrorMessages.defaultError.defaultMessage,
-            path: ['society'],
-          },
-        ),
-      withPeriod: z
-        .object({
-          dateFrom: z.string(),
-          dateTil: z.string(),
-        })
-        .refine(
-          ({ dateFrom, dateTil }) => new Date(dateFrom) <= new Date(dateTil),
+          ({ dateFrom, dateTil }) =>
+            dateFrom && dateTil
+              ? new Date(dateFrom) <= new Date(dateTil)
+              : true,
           {
             message: m.tilBeforeFrom.defaultMessage,
             path: ['dateTil'],
           },
         ),
+      place: z.object({
+        ceremonyPlace: z.string().optional(),
+        office: z.string().optional(),
+        society: z.string().optional(),
+      }),
     })
     .partial()
+    .refine(
+      ({ hasDate, period, date }) =>
+        hasDate === YES
+          ? (!!period?.dateFrom || !period?.dateFrom) &&
+            (!!period?.dateTil || !period?.dateTil) &&
+            !!date
+          : true,
+      {
+        message: coreErrorMessages.defaultError.defaultMessage,
+        path: ['date'],
+      },
+    )
+    .refine(
+      ({ hasDate, period, date }) =>
+        hasDate === NO
+          ? (!!date || !date) && !!period?.dateFrom && !!period.dateTil
+          : true,
```

Comment:
`(!!date || !date) === true`
```suggestion
        hasDate === NO && !!period?.dateFrom && !!period.dateTil,
```